### PR TITLE
(SIMP-8658) Update Docs for unpack_dvd

### PR DESCRIPTION
**Pull request by jeannegreulich**
- Updated use of unpack_dvd in General Upgrade instructions.
    - Updated the Extract Full ISO instructions and moved them to a HOWTO
      with new unpack_dvd instructions.
    - Added an example for updating an existing environment and linked to it
      from installation instructions.
    
    SIMP-8658 #close
    SIMP-8692 #close

### CODE REVIEW

**Comment by lnemsick-simp** [comment]
Did you want to tell the users to use the `--unpack-pxe --environment ENV` options when setting up the tftpboot server?

File: docs/user_guide/SIMP_Administration/Environments/Environments_Examples.inc [hunk 1]
```diff
@@ -23,6 +23,62 @@ You might also find the following HOWTO documents to be helpful:
...
+      # or a simp puppet module with a patch
+      yum update pupmod-simp-aide
+
+   This will update the local git repositories on the puppetserver.
+
+#. **Deploy the Environment**
+
+.. include:: ../common/Update_and_Deploy_Local_Environment.inc
+
+#. **Apply node configuration**
```
Comment (lnemsick-simp) [inline docs/user_guide/SIMP_Administration/Environments/Environments_Examples.inc:73]: numbering gets restarted to 1

File: docs/user_guide/SIMP_Administration/Environments/Environments_Examples.inc [hunk 2]
```diff
@@ -23,6 +23,62 @@ You might also find the following HOWTO documents to be helpful:
...
+   Then install or update any modules
+
+   .. code-block:: sh
+
+      # For example if installing all the modules in simp-extras
+      yum install simp-extras
+      # or a simp puppet module with a patch
+      yum update pupmod-simp-aide
+
+   This will update the local git repositories on the puppetserver.
```
Comment (lnemsick-simp) [inline docs/user_guide/SIMP_Administration/Environments/Environments_Examples.inc:67]: Should this be SIMP master or SIMP server (should be consistent)?  Line 48 above talks about SIMP master.

File: docs/user_guide/SIMP_Administration/Environments/Environments_Examples.inc [hunk 3]
```diff
@@ -23,6 +23,62 @@ You might also find the following HOWTO documents to be helpful:
...
+
+#. **Deploy the Environment**
+
+.. include:: ../common/Update_and_Deploy_Local_Environment.inc
+
+#. **Apply node configuration**
+
+   Any new modules will now be available to use in configurations on your nodes
+   in the updated envirments.
```
Comment (lnemsick-simp) [inline docs/user_guide/SIMP_Administration/Environments/Environments_Examples.inc:76]: 'envirments' => 'environments'

File: docs/user_guide/SIMP_Administration/Environments/Environments_Examples.inc [hunk 4]
```diff
@@ -23,6 +23,62 @@ You might also find the following HOWTO documents to be helpful:
...
+   Any changes in updated modules will be applied on the next scheduled  puppet run
+   on any node in the environment. Run :code:`puppet agent -t` to force puppet
```
Comment (lnemsick-simp) [inline docs/user_guide/SIMP_Administration/Environments/Environments_Examples.inc:78]: :code: -> :command:

File: docs/user_guide/Upgrade_SIMP/General_Upgrade_Instructions.rst [hunk 5]
```diff
@@ -73,25 +73,38 @@ steps as ``root``:
 
    * Update the repositories using a SIMP ISO:
 
-     If you have the latest SIMP ISO available to you and have installed the
-     ``simp-utils`` package, update the YUM repositories by unpacking the ISO
-     using ``unpack_dvd`` from that package:
+     :program:`unpack_dvd` can be used to extract the SIMP puppet module RPMs and the minimal OS
+     RPMs from the SIMP ISO. :program:`unpack_dvd` is installed from the :pupmod:`simp-utils` package.
```
Comment (lnemsick-simp) [inline docs/user_guide/Upgrade_SIMP/General_Upgrade_Instructions.rst:77]: :pupmod: => :package:

File: docs/user_guide/Upgrade_SIMP/General_Upgrade_Instructions.rst [hunk 6]
```diff
@@ -73,25 +73,38 @@ steps as ``root``:
...
 
-     #. Copy the new SIMP ISO file to the SIMP master
-     #. From the SIMP master (as ``root``):
+     By default :program:`unpack_dvd` uses information on the ISO to determine where to copy the RPMs
+     to under :file:`/var/www/yum` and then links the OS major version to the newly extracted OS directory.
+     Since sometimes :program:`unpack_dvd` can only determine the major version
+     of the OS you should supply a detailed version number for the OS using the -v option.
```
Comment (lnemsick-simp) [inline docs/user_guide/Upgrade_SIMP/General_Upgrade_Instructions.rst:82]: 'OS you' -> 'OS, you'

File: docs/user_guide/common/Update_and_Deploy_Local_Environment.inc [hunk 7]
```diff
@@ -1,53 +1,50 @@
 .. _ug-update-and-deploy-using-local-repos:
 
-#. Generate the new ``Puppetfile.simp``, a Puppetfile containing the latest
-   versions of only SIMP-packaged Puppet modules:
+  * Generate the Environment's Puppetfile.simp
 
-   .. code-block:: sh
+    Run :program:`simp` to pull all the latest versions of the SIMP-packaged Puppet modules
```
Comment (lnemsick-simp) [inline docs/user_guide/common/Update_and_Deploy_Local_Environment.inc:5]: Some places in this PR you use 'Puppet modules' and in others you use 'puppet modules'.  Don't know which is preferred, but would like to see some consistency.

File: docs/user_guide/common/Update_and_Deploy_Local_Environment.inc [hunk 8]
```diff
@@ -1,53 +1,50 @@
...
-.. TIP::
+       # The environment in this example is called `test`.  Replace `test` with the
+       # name of your environment.
+       ( umask 0027 && sg puppet -c '/usr/share/simp/bin/r10k puppetfile install \
+       --puppetfile /etc/puppetlabs/code/environments/test/Puppetfile \
+       --moduledir /etc/puppetlabs/code/environments/test/modules' )
 
-   Use ``simp --help`` for more information on the ``simp`` utility and alternate
-   options provided for each command.
+     Use the --force option if you get warnings that local changes will get
```
Comment (lnemsick-simp) [inline docs/user_guide/common/Update_and_Deploy_Local_Environment.inc:48]: --force ->  :command:`--force`.

**Review by lnemsick-simp — CHANGES_REQUESTED** [review]

File: docs/HOWTO/90_Misc/Unpack_dvd.rst [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+      # unpack_dvd -v <os version> <full path to iso>
+      unpack_dvd -v 7.8.2003 /myisodir/CentOS-7-x86_64-DVD-2003.iso
+
+
+#. Ensure that subsequent :term:`yum` operations are aware of the new RPM
+   packages by refreshing the system's yum cache:
+
+   .. code:: bash
+
+      yum clean all && yum makecache
```
[the remark this answers is not in the document]
Comment (jeannegreulich) [reply docs/HOWTO/90_Misc/Unpack_dvd.rst:39]: it sets the permissions to root:apache.

File: docs/HOWTO/90_Misc/Unpack_dvd.rst [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+.. NOTE::
+
+   If the OS version in the :file:`.treeinfo`  is just the major version, as it is on
+   most CentOS ISOs, :program:`unpack_dvd` will ask you to supply a more descriptive
+   version number using ``-v`` option.
+
+#. Log on as ``simp`` and run :command:`sudo su - root`.
+#. Copy the appropriate vendor OS ISO(s) to the server.
+#. If the server where you are unpacking the vendor ISO was **NOT** built using the SIMP ISO ,
+   you must create a SIMP directory in :file:`/var/www/yum/` (or the directory you indicated in ``-d``
```
[the remark this answers is not in the document]
Comment (jeannegreulich) [reply docs/HOWTO/90_Misc/Unpack_dvd.rst]: I tested again.  I must not have been using the latest version last time.  Fixed.

File: docs/HOWTO/90_Misc/Unpack_dvd.rst [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+the :program:`unpack_dvd`.
+
+The :program:`unpack_dvd` extracts the OS rpms to :file:`/var/www/yum/<OperatingSystem>`
+in a directory named after the OS version in the :file:`.treeinfo` file on the DVD.
+It then creates a link to the major version.  If version 7.6 is extracted, 
+7 will be linked to 7.6.
+
+.. NOTE::
+
+   If the OS version in the :file:`.treeinfo`  is just the major version, as it is on
```
[the remark this answers is not in the document]
Comment (jeannegreulich) [reply docs/HOWTO/90_Misc/Unpack_dvd.rst]: fixed.

File: docs/user_guide/SIMP_Administration/Environments/Environments_Examples.inc (hunk 2 above)
Comment (jeannegreulich) [reply docs/user_guide/SIMP_Administration/Environments/Environments_Examples.inc:67]: fixed

File: docs/user_guide/SIMP_Administration/Environments/Environments_Examples.inc (hunk 4 above)
Comment (jeannegreulich) [reply docs/user_guide/SIMP_Administration/Environments/Environments_Examples.inc:78]: I don't like the way command looks.  It is this big black blob.  I grepped and command is used alotso I guess if we want we can just change it in the defnition. Fixed.

File: docs/HOWTO/90_Misc/Unpack_dvd.rst (hunk 9 above)
[the remark this answers is not in the document]
Comment (lnemsick-simp) [reply docs/HOWTO/90_Misc/Unpack_dvd.rst:39]: It did not set the permissions on the system I used for testing (based on the simp-core PR).  Could you please recheck this?

File: docs/HOWTO/90_Misc/Unpack_dvd.rst [hunk 12]
```diff
@@ -38,11 +53,47 @@ It then creates a link to the major version.  If version 7.6 is extracted,
...
+* extract the PXE files to :file:`/my/tftpboot/`
+
+.. code:: bash
+
+   # The PXE directory must follow the -X option.
+   unpack_dvd -v 8.0.1905 -d /my/repodir /myisodir/CentOS--x86_64-1905-dvd1.iso -X /my/tftpboot
+
+.. NOTE::
+
+   You cannot just extract the PXE files.  It will always extract the packages
```
Comment (lnemsick-simp) [inline docs/HOWTO/90_Misc/Unpack_dvd.rst:103]: There is a `--no-unpack-yum` option to exclude package extraction.  Does this not work?

File: docs/HOWTO/90_Misc/Unpack_dvd.rst [hunk 13]
```diff
@@ -1,28 +1,43 @@
...
-=========================================
+HOWTO Use unpack_dvd to Extract the Full OS RPM Package Set
+===========================================================
 
 The SIMP ISO provides a minimal set of packages.
 
 If you require additional OS packages, you can extract them from vendor ISOs using
-:program:`unpack_dvd`.
+:program:`unpack_dvd`.  Additionally you can extract the extract the PXE boot
+TFTP boot files from the ISO to the rsync directories.
```
Comment (lnemsick-simp) [inline docs/HOWTO/90_Misc/Unpack_dvd.rst:10]: the TFTP boot files extracted are the kernel and root filesystem files.  They are distinct from the boot files (menu.c32, pxelinux.0, grub.efi, shim.efi etc)

File: docs/HOWTO/90_Misc/Unpack_dvd.rst [hunk 14]
```diff
@@ -1,28 +1,43 @@
...
+
 .. NOTE::
 
    If unpack_dvd can only determine the major OS version from the files
    on the ISO, :program:`unpack_dvd` will ask you to supply a more descriptive
    version number using ``-v`` option.
 
+The following example will:
+
+* extract the RPMs to :file:`/var/www/yum/CentOS/7.8.2003`
```
Comment (lnemsick-simp) [inline docs/HOWTO/90_Misc/Unpack_dvd.rst:33]: When I ran the program the files were root:root 0644.  If they should be root:apache 0640, the user will have to adjust.

File: docs/HOWTO/90_Misc/Unpack_dvd.rst [hunk 15]
```diff
@@ -0,0 +1,105 @@
+.. _howto-unpack-dvd:
+
+HOWTO Use unpack_dvd to Extract the Full OS RPM Package Set
+===========================================================
+
+The SIMP ISO provides a minimal set of packages.
+
+If you require additional OS packages, you can extract them from vendor ISOs using
+:program:`unpack_dvd`.  Additionally you can extract the extract the PXE boot
```
Comment (lnemsick-simp) [inline docs/HOWTO/90_Misc/Unpack_dvd.rst:9]: 'extract the extract' => 'extract'

File: docs/HOWTO/90_Misc/Unpack_dvd.rst [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+   on the ISO, :program:`unpack_dvd` will ask you to supply a more descriptive
+   version number using ``-v`` option.
+
+The following example will:
+
+* extract the RPMs to :file:`/var/www/yum/CentOS/7.8.2003`
+* create a repository in the above directory
+* link :file:`/var/www/yum/CentOS/7` to the above directory.
+
+#. Log on as ``simp`` and run :command:`sudo su - root`.
```
Comment (lnemsick-simp) [inline docs/HOWTO/90_Misc/Unpack_dvd.rst:37]: If you install SIMP from RPM, there will not be a `simp` user.  How about just telling the user to login as root?

File: docs/HOWTO/90_Misc/Unpack_dvd.rst [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+   At this time :program:`unpack_dvd` does not work entirely with EL8 ISOs.
+   EL8 introduced modules to repositories and :program:`unpack_dvd` can not handle these.
+   It will extract some but not all of RPMs on an EL8 ISO. The files extracted
+   are enough to kickstart a basic EL8 system but some of the application stream
+   files are not extracted and added to the repository correctly.
+
+Extract PXE files
+-----------------
+
+Extracting the PXE files along with the OS files, was added to :program:`unpack_dvd` in :package:`simp-utils-6.4.0`.
```
Comment (lnemsick-simp) [inline docs/HOWTO/90_Misc/Unpack_dvd.rst:67]: 'Extracting the PXE files along with the OS files,' -> "The ability to extract the PXE files'

File: docs/HOWTO/90_Misc/Unpack_dvd.rst [hunk 18]
```diff
@@ -0,0 +1,105 @@
...
+
+By default :program:`unpack_dvd` will pull information off the ISO and, using this information, create a directory named <os-family>-<version>-<arch>  under the tftpboot rsync directory and extract the PXE files there.
```
Comment (lnemsick-simp) [inline docs/HOWTO/90_Misc/Unpack_dvd.rst:69]: Suggested replacment:
When PXE boot file extraction is enabled with the :command:`-X` option,
:program:`unpack_dvd` will, by default,

* extract OS version information from the ISO metadata
* create a directory for the PXE boot files in the rsync directory used by the
  :pupmod:`simp/tftpboot` module for the `production` Puppet environment

  * The new directory will be named :file:`<os-family>-<version>-<arch>`.
  * The `production` rsync directory is located at
    :file:`/var/simp/environments/production/rsync/<os family>/Global/tftpboot/linux-install/`.
  * See :ref:`Configure_PXE_Boot` for more information about using
    :pupmod:`simp/tftpboot`.

* extract the PXE files to the created OS directory.

File: docs/HOWTO/90_Misc/Unpack_dvd.rst [hunk 19]
```diff
@@ -0,0 +1,105 @@
...
+* extract the RPMs to :file:`/var/www/yum/CentOS/8.0.1905`
+* create a repository in the above directory
+* link :file:`/var/www/yum/CentOS/8` to the above directory
+* extract the PXE files to :file:`/var/simp/environments/test/rsync/CentOS/Global/tftpboot/linux-install/centos-8.0.1905-x86_64`
+* link `/var/simp/environments/test/rsync/CentOS/Global/tftpboot/linux-install/centos-8-x86_64` to the above directory.
+
+
+.. code:: bash
+
+   # Place the -X options after the ISO name
```
Comment (lnemsick-simp) [inline docs/HOWTO/90_Misc/Unpack_dvd.rst:86]: Why must -X be after the ISO name?  Shouldn't matter.

File: docs/HOWTO/90_Misc/Unpack_dvd.rst [hunk 20]
```diff
@@ -0,0 +1,105 @@
...
+The following example will
+
+* extract the RPMs  to :file:`/my/repodir/yum/CentOS/8.0.1905`
+* create a repository in the above directory
+* extract the PXE files to :file:`/my/tftpboot/`
+
+.. code:: bash
+
+   # The PXE directory must follow the -X option.
+   # Use the -n to prevent the creation of the links.
```
Comment (lnemsick-simp) [inline docs/HOWTO/90_Misc/Unpack_dvd.rst:98]: -n was used in the previous example.  Is this to indicate something else they can do or was the comment intended for the previous example.

File: docs/HOWTO/90_Misc/Unpack_dvd.rst [hunk 21]
```diff
@@ -0,0 +1,105 @@
...
+   files are not extracted and added to the repository correctly.
+
+Extract PXE files
+-----------------
+
+Extracting the PXE files along with the OS files, was added to :program:`unpack_dvd` in :package:`simp-utils-6.4.0`.
+
+By default :program:`unpack_dvd` will pull information off the ISO and, using this information, create a directory named <os-family>-<version>-<arch>  under the tftpboot rsync directory and extract the PXE files there.
+
+The default rsync directory is :file:`/var/simp/environments/production/rsync/<os family>/Global/tftpboot/linux-install/`.  Options exist to change the environment in the rsync directory or to specify an alternate directory.
```
Comment (lnemsick-simp) [inline docs/HOWTO/90_Misc/Unpack_dvd.rst:71]: Suggested replacement:
You can change the Puppet environment or the root destination directory for the PXE files with other :program:`unpack_dvd` options.  Just be sure the root destination directory exists before running :program:`unpack_dvd`.

File: docs/HOWTO/90_Misc/Unpack_dvd.rst (hunk 14 above)
Comment (jeannegreulich) [reply docs/HOWTO/90_Misc/Unpack_dvd.rst:33]: I found that but it is an error introduced.  I am looking to see if I can fix it.

Comment (jeannegreulich) [reply docs/HOWTO/90_Misc/Unpack_dvd.rst:33]: It does change the permissions.  I changes the permissions only of the repos which is all they need.  I made some notes in the HOWTO about permissions.

**Review by lnemsick-simp — APPROVED** [review]

**Pull request merged**